mintapi/api.py

```diff
@@ -616,75 +616,23 @@ def populate_extended_account_detail(self, accounts):  # {{{
 
         return accounts
 
-    def get_budgets(self, hist=None):
-        response = self.__call_budgets_endpoint()
-        categories = self.get_categories()
-        income = response["data"]["income"]
-        spending = response["data"]["spending"]
-        if hist is not None:  # version proofing api
-
-            def mos_to_yrmo(mos_frm_zero):
-                return datetime(
-                    year=int(mos_frm_zero / 12), month=mos_frm_zero % 12 + 1, day=1
-                ).strftime("%Y%m")
-
-            # Error checking 'hist' argument
-            if isinstance(hist, str) or hist > 12:
-                hist = 12  # MINT_ROOT_URL only calls last 12 months of budget data
-            elif hist < 1:
-                hist = 1
-
-            bgt_cur_mo = max(map(int, income.keys()))
-            min_mo_hist = bgt_cur_mo - hist
-
-            # Initialize and populate dictionary for return
-            #   Output 'budgets' dictionary with structure
-            #       { "YYYYMM": {"spending": [{"key": value, ...}, ...],
-            #                      "income": [{"key": value, ...}, ...] } }
-            budgets = {}
-            for months in range(bgt_cur_mo, min_mo_hist, -1):
-                budgets[mos_to_yrmo(months)] = {}
-                budgets[mos_to_yrmo(months)]["income"] = income[str(months)]["bu"]
-                budgets[mos_to_yrmo(months)]["spending"] = spending[str(months)]["bu"]
-
-            # Fill in the return structure
-            for month in budgets.keys():
-                for direction in budgets[month]:
-                    for budget in budgets[month][direction]:
-                        budget = self.__format_budget_categories(budget, categories)
-
+    def get_budgets(self):
+        budgets = self.__call_budgets_endpoint()
+        if "Budget" in budgets.keys():
+            for i in budgets["Budget"]:
+                i["lastUpdatedDate"] = i["metaData"]["lastUpdatedDate"]
+                i.pop("metaData", None)
         else:
-            # Make the skeleton return structure
-            budgets = {
-                "income": income[str(max(map(int, income.keys())))]["bu"],
-                "spend": spending[str(max(map(int, spending.keys())))]["bu"],
-            }
-
-            # Fill in the return structure
-            for direction in budgets.keys():
-                for budget in budgets[direction]:
-                    budget = self.__format_budget_categories(budget, categories)
-
-        return budgets
+            raise MintException("Cannot find budget data")
```

Comment:
@mrooney Good point.  Since this is consistent with error messaging with Category and Investment so far, I'd say let's leave this as is for now and then do a follow-up PR that handles more specific error handling from the new endpoints.